tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -688,23 +703,13 @@ extern k_thread_stack_t ztest_thread_stack[];
 void test_main(void)
 {
 	/* partitions must be initialized first */
-	FOR_EACH(appmem_init_part, part0, part1, part2);
+	APPMEM_INIT_PARTITIONS(part0, part1, part2);
 	/*
 	 * Next, the app_memory must be initialized in order to
 	 * calculate size of the dynamically created subsections.
 	 */
-#if defined(CONFIG_ARC)
```

Comment:
This change looks "un-related" to the pull-request, IMHO. This should be submitted separately and get ARC guys review it and confirm that such patches are not needed; i.e. supervisors have access to configure and initialize memory without need for disabling and re-enabling mpu...